spec/factories.rb

```diff
@@ -410,7 +422,18 @@
   end
 
   factory :line_item_with_shipment, parent: :line_item do
-    target_shipment { create(:shipment, order: order) }
+    transient do
+      shipping_fee 3
+    end
+
+    target_shipment do
+      shipment = order.reload.shipments.first
+      if shipment == nil
```

Comment:
the idiomatic way would be 
```suggestion
      if shipment.nil?
```

Reply:
Thanks a lot!!! Learning Ruby bit by bit :-)
I cant believe I couldn't write a line of Ruby 6 months ago, I can surely READ ruby fluently now :-D